odoo/orm/fields_properties.py

```diff
@@ -29,6 +29,79 @@ def check_property_field_value_name(property_name):
     if not (0 < len(property_name) <= 512) or not regex_alphanumeric.match(property_name):
         raise ValueError(f"Wrong property field value name {property_name!r}.")
 
+class Property(dict):
+    """Allow to be used in `convert_to_record` to make the verification only if we try accessing the property value."""
```

Comment:
That's gibberish to me.  Please describe what a `Property` object is.  If I am not mistaken, it is the value of a "properties" field in the "record" format, i.e., the result of evaluating `record.property_field`.

Reply:
Yes, it's the result of `record.property_field`, the thing is that you can write a mail template, inserting a property of the "CRM team A", but execute that template on a record of the "CRM team B", this shouldn't crash imo, like for standard field

Once the template is validated, it shouldn't crash